File: back-End/src/main/java/com/spring/student/studentsystem/model/Student.java
package com.spring.student.studentsystem.model;

import java.util.Date;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import com.spring.student.studentsystem.model.enums.Gender;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "student")
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    
    @Column(name = "id")
    private Long id;
    @Column(name = "Full_name")
    private String fullName;
    @Enumerated(EnumType.STRING)
    @Column(name = "gender")
    private Gender gender;
    @Column(name = "phone_number")
    private String phone;
    @Column(name = "age")
    private String age;
    @Column(name = "address")
    private String address;
    @CreationTimestamp
    @Column(name = "date_created")
    private Date dateCreated;
    @UpdateTimestamp
    @Column(name = "date_updated")
    private Date dateUpdated;
 
}

File: front-End/src/app/model/student.ts
export class Student {
    id: number;
    fullName: string;
    gender: string;
    age: string;
    phone: string;
    address: string;
    dateCreated: Date;
    dateUpdated: Date;
    constructor(id: number, fullName: string, age: string, address: string, phone: string, gender: string) {
      this.id = id;
      this.fullName = fullName;
      this.gender = gender;
      this.age = age;
      this.phone = phone;
      this.address = address;
    }

  }